src/sentry/static/sentry/app/views/settings/projectSecurityHeaders/index.jsx

```diff
@@ -0,0 +1,101 @@
+import {Box, Flex} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import {t} from '../../../locale';
+import AsyncView from '../../asyncView';
+import Button from '../../../components/buttons/button';
+import {Panel, PanelBody, PanelHeader, PanelItem} from '../../../components/panels';
+import ReportURI from './reportUri';
+import PreviewFeature from '../../../components/previewFeature';
+import SettingsPageHeader from '../components/settingsPageHeader';
+import TextBlock from '../components/text/textBlock';
+
+const HeaderName = styled.span`
+  font-size: 1.2em;
+`;
+
+export default class ProjectSecurityHeaders extends AsyncView {
+  static propTypes = {
+    setProjectNavSection: PropTypes.func,
+  };
+
+  componentWillMount() {
+    super.componentWillMount();
+    this.props.setProjectNavSection('settings');
+  }
+
+  getEndpoints() {
+    let {orgId, projectId} = this.props.params;
+    return [['keyList', `/projects/${orgId}/${projectId}/keys/`]];
+  }
+
+  getReports() {
+    let {orgId, projectId} = this.props.params;
+    return [
+      {
+        name: 'Content Security Policy',
+        description: 'foo bar',
+        url: `/settings/${orgId}/${projectId}/security-headers/csp/`,
```

Comment:
if this is accessed in old settings, they'll be forwarded to new settings

Reply:
how much do we care? what is the other solution?

Reply:
You could try `recreateRoute('csp/', this.props)`